server/game_engine/GameEngine.js

```diff
@@ -145,25 +146,117 @@ class Game {
     }
 
     shuffleBoard() {
-        let words = ["Cat", "Dog", "Bird", "Fox", "Monkey", "Snake", "Panda", "Dinosaur", "Dolphin", "Human", "Monster", "Slime",
-            "Blueberry", "Strawberry", "Orange", "Mango", "Banana", "Apple", "Tomato", "Cucumber", "Cherry", "Avocado",
-            "Car", "Airplane", "Bike", "Truck", "Tesla",
-            "Sword", "Shield", "Staff", "Bow", "Helmet", "Dagger", "Gun", "Belt", "Armor",
-            "Beach", "Jungle", "Desert", "Water", "Earth", "Wind", "Fire", "Grass", "Space", "Snow", "Moon", "Electricity",
-            "Ball", "Tail", "Shoe", "Rainbow", "Pole", "Computer", "Cellphone", "Camera", "Bitcoin", "Money", "Book",
-            "Television", "House", "Doll",
-            "Run", "Change", "Teleport", "Slash", "Switch", "Eat", "Picture", "Dare", "Retire"];
-        let colors = ["blue", "blue", "blue", "blue", "blue", "blue", "blue", "blue", "blue",
-            "red", "red", "red", "red", "red", "red", "red", "red",
-            "white", "white", "white", "white", "white", "white", "white",
-            "black"]; //Counter for card colors to be distributed (9 blue, 8 red, 7 white, 1 black)
+        let words = [
+            "Cat",
+            "Dog",
+            "Bird",
+            "Fox",
+            "Monkey",
+            "Snake",
+            "Panda",
+            "Dinosaur",
+            "Dolphin",
+            "Human",
+            "Monster",
+            "Slime",
+            "Blueberry",
+            "Strawberry",
+            "Orange",
+            "Mango",
+            "Banana",
+            "Apple",
+            "Tomato",
+            "Cucumber",
+            "Cherry",
+            "Avocado",
+            "Car",
+            "Airplane",
+            "Bike",
+            "Truck",
+            "Tesla",
+            "Sword",
+            "Shield",
+            "Staff",
+            "Bow",
+            "Helmet",
+            "Dagger",
+            "Gun",
+            "Belt",
+            "Armor",
+            "Beach",
+            "Jungle",
+            "Desert",
+            "Water",
+            "Earth",
+            "Wind",
+            "Fire",
+            "Grass",
+            "Space",
+            "Snow",
+            "Moon",
+            "Electricity",
+            "Ball",
+            "Tail",
+            "Shoe",
+            "Rainbow",
+            "Pole",
+            "Computer",
+            "Cellphone",
+            "Camera",
+            "Bitcoin",
+            "Money",
+            "Book",
+            "Television",
+            "House",
+            "Doll",
+            "Run",
+            "Change",
+            "Teleport",
+            "Slash",
+            "Switch",
+            "Eat",
+            "Picture",
+            "Dare",
+            "Retire",
+        ];
+        let colors = [
```

Comment:
Maybe we can create this more programatically, for example the finite set of different elements are "blue", "red", "white" and "black"

Populate the data 9 blues, 8 reds, 7 whites, 1 black in for loops

Reply:
If I understand correctly I did have something like this in one of the earlier versions of the game engine. I had an array of length 4 (one index for each color) and assign cards a color based on one of the indices (capping at the correct amount). The problem I had was that each color had an equal (25%) chance of getting assigned so the black card was almost always near the top left of the board. I would need to add probabilities which also change after each card gets assigned. I was planning on looking at this again later on when the important stuff gets finished and put this as a quick fix.

Reply:
Sounds good! I've created a ticket to be addressed at a later date https://github.com/hatchways/team-iron-man/issues/67